backend/controllers/cellMove/cellMoveViewResidentialLocation.js

```diff
@@ -0,0 +1,84 @@
+const { alertFlagLabels, cellMoveAlertCodes } = require('../../shared/alertFlagValues')
+const { putLastNameFirst, formatLocation } = require('../../utils')
+
+const prisonApiLocationId = async (res, whereaboutsApi, locationKey, userCaseLoad) => {
```

Comment:
I decided to do the location-description-building here as the `whereabouts-api` also returns sub-locations. For those sub-locations there is no obvious, reliable way of generating the location description.